packages/widget/src/hooks/useChains.ts

```diff
@@ -17,34 +17,36 @@ export const useChains = () => {
       staleTime: 300000,
     });
 
-  const { data: filteredChains, isLoading: isLoadingFilteredChains } = useQuery(
-    ['filtered-chains', availableChains?.length, keyPrefix],
-    async () => {
-      if (!availableChains) {
-        return;
-      }
-      const filteredChains = availableChains.filter((chain) =>
-        isItemAllowed(chain.id, chains),
-      );
-      const chainOrder = initializeChains(
-        filteredChains.map((chain) => chain.id),
-      );
-      const [fromChainValue, toChainValue] = getValues([
-        FormKey.FromChain,
-        FormKey.ToChain,
-      ]);
-      if (!fromChainValue) {
-        setValue(FormKey.FromChain, chainOrder[0]);
-      }
-      if (!toChainValue) {
-        setValue(FormKey.ToChain, chainOrder[0]);
-      }
-      return filteredChains;
-    },
-    {
-      enabled: Boolean(availableChains),
-    },
-  );
+  const filteredChains = useMemo(() => {
+    if (!availableChains) {
+      return;
+    }
+
+    const filteredAvailableChains = availableChains.filter((chain) =>
+      isItemAllowed(chain.id, chains),
+    );
+    const chainOrder = initializeChains(
+      filteredAvailableChains.map((chain) => chain.id),
+    );
+    const [fromChainValue, toChainValue] = getValues([
+      FormKey.FromChain,
+      FormKey.ToChain,
+    ]);
+    if (!fromChainValue) {
+      setValue(FormKey.FromChain, chainOrder[0]);
+    }
+    if (!toChainValue) {
+      setValue(FormKey.ToChain, chainOrder[0]);
+    }
+    return filteredAvailableChains;
+  }, [
+    keyPrefix,
```

Comment:
Could do with double checking and understanding the use of keyPrefix here too, I've added it to the dep array as it was used in the queryKey previously

Reply:
keyPrefix is used in the config for when you might have multiple instance of the widget running in the same app